tests/integration/devfile/cmd_devfile_debug_test.go

```diff
@@ -0,0 +1,171 @@
+package devfile
+
+import (
+	"github.com/openshift/odo/pkg/util"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/openshift/odo/tests/helper"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("odo devfile debug command tests", func() {
+	var namespace, context, componentName, currentWorkingDirectory, projectDirPath string
+	var projectDir = "/projectDir"
+
+	// This is run after every Spec (It)
+	var _ = BeforeEach(func() {
+		SetDefaultEventuallyTimeout(10 * time.Minute)
+		SetDefaultConsistentlyDuration(30 * time.Second)
+		namespace = helper.CreateRandProject()
+		context = helper.CreateNewContext()
+		currentWorkingDirectory = helper.Getwd()
+		projectDirPath = context + projectDir
+		componentName = helper.RandString(6)
+
+		helper.Chdir(context)
+
+		os.Setenv("GLOBALODOCONFIG", filepath.Join(context, "config.yaml"))
+
+		// Devfile push requires experimental mode to be set
+		helper.CmdShouldPass("odo", "preference", "set", "Experimental", "true")
+	})
+
+	// Clean up after the test
+	// This is run after every Spec (It)
+	var _ = AfterEach(func() {
+		helper.DeleteProject(namespace)
+		helper.Chdir(currentWorkingDirectory)
+		helper.DeleteDir(context)
+		os.Unsetenv("GLOBALODOCONFIG")
+	})
+
+	Context("odo debug on a nodejs:latest component", func() {
```

Comment:
I can see only nodejs devfile component, Can you please add the same verification for java component too. use `odo catalog list components` in experimental mode to check list of supported devfile component.

Reply:
The supported devfile components don't contain the debug step in the devfile and I haven't found any devfile with it either till now. Let me search some other repos.

Reply:
you can add debug step information into the devfile registry in https://github.com/elsony/devfile-registry and https://github.com/odo-devfiles/registry

Reply:
But I have no idea how to start remote debugging for java components. Let me search for a example.

Reply:
I did manage to attach the debugger to a java component and stop at a breakpoint. 

But the problem is, the way we verify the debugger attachment in case of nodejs components, doesn't work for java components. In case of nodejs components, hitting localhost:<port> returns a string like `web-socket` connection was expected` but in case of java we get a empty response.

Reply:
After a discussion with @kadel , we have decided to handle it in a separate PR since none of our debug tests are currently are based on Java. This PR already has the necessary steps to verify that debugging works from odo's side for devfiles.

 We have opened a new issue https://github.com/openshift/odo/issues/3378.

Reply:
> After a discussion with @kadel , we have decided to handle it in a separate PR since none of our debug tests are currently are based on Java.

Make sense.
I have updated in the issue https://github.com/openshift/odo/issues/2153